PubHub/PubHub.API/Domain/Auth/WhitelistService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Options;
+using PubHub.API.Controllers.Problems;
+
+namespace PubHub.API.Domain.Auth
+{
+    public class WhitelistService
+    {
+        private readonly WhitelistOptions _whitelistOptions;
+        private readonly Func<IResult> _unauthorizedResult = () => Results.Problem(statusCode: UnauthorizedSpecification.STATUS_CODE, title: UnauthorizedSpecification.TITLE, detail: "Unauthorized access to resource.");
+
+        public WhitelistService(IOptions<WhitelistOptions> whitelistOptions)
+        { 
+            _whitelistOptions = whitelistOptions.Value;
+        }
+
+        /// <summary>
+        /// Verify whether an application is allowed to access a given controller and endpoint.
+        /// </summary>
+        /// <param name="appId">Unique and secret application ID of an application accessing the PubHub API.</param>
+        /// <param name="controllerName">Name of the controller trying to be accessed.</param>
+        /// <param name="methodName">Name of controller method trying to be accessed.</param>
+        /// <param name="problem">Null when returning true; otherwise problem from <see cref="UnauthorizedSpecification"/>.</param>
+        /// <returns>True if the application as access; otherwise false.</returns>
```

Comment:
Minor thing. Don't have to change it, however, it' possible to highlight `true` and `false` by writing `<see langword="true"/>` and `<see langword="false"/>`. Just for info (:

Also there's a spelling error on lime 25. Should be _has_ insead of _as_

Reply:
Oh, I wasn't aware of `langword`. I'll try to use that and since I'll be changing other things for this PR, I'll try it out right now. 😉
The typo should also be fixed.